src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -26,56 +28,44 @@
 
 Data by USGS is published quickly after the dekad.
 After about 1 month this data is updated with temporal smoothing
-and error correction for cloud cover.
+and error correction for cloud cover. Files for a specific
+dekad and region can range from 30MB up to over 100MB, so
+downloading and processing can take a long time.
 """
 
 # TODO: add progress bar
 import logging
+import re
 from datetime import date
 from io import BytesIO
 from pathlib import Path
-from typing import List, Tuple, Union, cast
+from typing import List, Optional, Tuple, Union
 from urllib.error import HTTPError
 from urllib.request import urlopen
 from zipfile import ZipFile
 
 import geopandas as gpd
 import pandas as pd
-import regex as re
 import rioxarray  # noqa: F401
 import xarray as xr
-from typing_extensions import Literal
 
 import aatoolbox.utils.raster  # noqa: F401
 from aatoolbox.config.countryconfig import CountryConfig
 from aatoolbox.datasources.datasource import DataSource
-from aatoolbox.utils._dates import (
+from aatoolbox.utils.dates import (
     _compare_dekads_gt,
     _compare_dekads_lt,
-    _date_to_dekad,
     _dekad_to_date,
     _expand_dekads,
+    _get_dekadal_date,
 )
 
-# TODO: use check_file_existence once the fix is merged
-# that allows it to work on class methods (PR is open)
-# from aatoolbox.utils.io import check_file_existence
+# from aatoolbox.utils.check_file_existence import check_file_existence
```

Comment:
why is this and the decorator at `_download` commented out?

Reply:
I am having trouble at the moment rebasing to develop even though these functions are now in there. Not sure why. Trying to sort.

Reply:
Had [this issue](https://stackoverflow.com/questions/56636172/why-is-git-rebase-flagging-conflicts-on-new-branch-specific-files) with `git rebase develop` flagging new files repeatedly for conflicts. The SO post helped fix it.

Reply:
No longer using `check_file_existence` anyway, though, as we are doing much more complex checks based on modification time.